### PR TITLE
make polygon specific provider / signer calls dynamic to the network

### PR DESCRIPTION
**Pull request by herbig**
## Description

I took what was done in https://github.com/decent-dao/fractal-interface/pull/999 and put those calls into a method that delineates the calls depending on which network you're on.

This way we can bring back polygon, but not rely on the provider for other networks.

### CODE REVIEW

File: src/helpers/crypto.ts [hunk 1]
```diff
@@ -188,3 +190,15 @@ export const buildMultiSendSafeTx = (
 ): SafeTransaction => {
   return buildContractCall(multiSend, 'multiSend', [encodeMultiSend(txs)], nonce, true, overrides);
 };
+
+/**
+ * TODO explain why exactly we're doing this...
+ */
+export function asProper<T>(connection: ContractConnection<T>, chainId: number): T {
```
Comment (herbig) [inline src/helpers/crypto.ts:197]: this is the main point of the PR, every other change is just implementing everywhere that was changed in https://github.com/decent-dao/fractal-interface/pull/999 and filtering the chainId to it

Comment (herbig) [reply src/helpers/crypto.ts:197]: @Da-Colon I could use a better understanding of what the issue is though, to document it here.  Something to do with polygon block times? We'll discuss tomorrow.

File: src/helpers/crypto.ts [hunk 2]
```diff
@@ -188,3 +190,15 @@ export const buildMultiSendSafeTx = (
 ): SafeTransaction => {
   return buildContractCall(multiSend, 'multiSend', [encodeMultiSend(txs)], nonce, true, overrides);
 };
+
+/**
+ * TODO explain why exactly we're doing this...
+ */
+export function eventRPC<T>(connection: ContractConnection<T>, chainId: number): T {
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/helpers/crypto.ts:197]: yeah good call

File: src/helpers/crypto.ts [hunk 3]
```diff
@@ -188,3 +190,15 @@ export const buildMultiSendSafeTx = (
 ): SafeTransaction => {
   return buildContractCall(multiSend, 'multiSend', [encodeMultiSend(txs)], nonce, true, overrides);
 };
+
+/**
+ * TODO explain why exactly we're doing this...
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/helpers/crypto.ts:195]: Yeah,  I can include it here in the PR.

What's the technical explanation for why we use this for events on polygon?  The block times are faster, but why does it matter whether its provider vs. signer?

**Review by Da-Colon — APPROVED** [review]
Good call, makes sense

**Comment by herbig** [comment]
@decent-dao/fractal I think this logic is still good for supporting Polygon post Eth denver, but given that we're rewriting the contracts and invalidate existing DAOs, I'm going to still remove Polygon from the supported networks

**Pull request merged**